### PR TITLE
Fix typos in docs, code, and notebooks

### PR DESCRIPTION
**Pull request by nikosavola**
This PR fixes typos in docs, notebooks, code description, comments and some varibale names

### CODE REVIEW

**Review by gcarleo — APPROVED** [review]
wow thanks for fixing all these! much appreciated

File: docs/docs/sampler.ipynb [hunk 1]
```diff
@@ -352,14 +352,14 @@
...
+    "    possible couples (clusters). To construct it, the user must provide a graph object or a list of edges. **This sampler does respect constraints on Hilbert spaces, but might not explore the whole hilbert space because it preserves the total number of excitations/magnetization.**\n",
     " - {class}`rules.HamiltonianRule`: which transitions the configuration according to the off-diagonal elements of the Hamiltonian. As this rule might not be symmetric, this uses the factor $L(s,s')$ to reweight the transition probability.\n",
     " \n",
     "For Continuous Hilbert spaces, the only rule that we implement (at the moment is the following:\n",
-    " - {class}`rules.GaussianRule` which proposes a move by adding a normally-distributed random number to a single degree of freedom. The standard deviation of the guassian distribution can be modified.\n",
+    " - {class}`rules.GaussianRule` which proposes a move by adding a normally-distributed random number to a single degree of freedom. The standard deviation of the gaussian distribution can be modified.\n",
```
Comment (femtobit) [inline docs/docs/sampler.ipynb:362]:
```suggestion
    " - {class}`rules.GaussianRule` which proposes a move by adding a normally-distributed random number to a single degree of freedom. The standard deviation of the Gaussian distribution can be modified.\n",
```

File: docs/tutorials/gs-ising.ipynb [hunk 2]
```diff
@@ -1037,7 +1037,7 @@
    "id": "26014bdc-f72b-466f-b7d7-761a7499ce25",
    "metadata": {},
    "source": [
-    "And we can compare the results between the two different ansatze:"
+    "And we can compare the results between the two different Änsatze:"
```
Comment (femtobit) [inline docs/tutorials/gs-ising.ipynb:1040]:
```suggestion
    "And we can compare the results between the two different Ansätze:"
```

File: netket/optimizer/qgt/qgt_jacobian_pytree.py [hunk 3]
```diff
@@ -141,11 +141,11 @@ class QGTJacobianPyTreeT(LinearOperator):
 
     mode: str = struct.field(pytree_node=False, default=Uninitialized)
     """Differentiation mode:
-        - "real": for real-valued R->R and C->R ansatze, splits the complex inputs
+        - "real": for real-valued R->R and C->R Änsatze, splits the complex inputs
```
Comment (femtobit) [inline netket/optimizer/qgt/qgt_jacobian_pytree.py:144]:
```suggestion
        - "real": for real-valued R->R and C->R Ansätze, splits the complex inputs
```

File: netket/optimizer/qgt/qgt_jacobian_pytree.py [hunk 4]
```diff
@@ -141,11 +141,11 @@ class QGTJacobianPyTreeT(LinearOperator):
...
                   into real and imaginary part.
-        - "complex": for complex-valued R->C and C->C ansatze, splits the complex
+        - "complex": for complex-valued R->C and C->C Änsatze, splits the complex
```
Comment (femtobit) [inline netket/optimizer/qgt/qgt_jacobian_pytree.py:146]:
```suggestion
        - "complex": for complex-valued R->C and C->C Ansätze, splits the complex
```

File: netket/optimizer/qgt/qgt_jacobian_pytree.py [hunk 5]
```diff
@@ -141,11 +141,11 @@ class QGTJacobianPyTreeT(LinearOperator):
...
                   inputs and outputs into real and imaginary part
-        - "holomorphic": for any ansatze. Does not split complex values.
+        - "holomorphic": for any Änsatze. Does not split complex values.
```
Comment (femtobit) [inline netket/optimizer/qgt/qgt_jacobian_pytree.py:148]:
```suggestion
        - "holomorphic": for any Ansätze. Does not split complex values.
```

File: docs/advanced/custom_models.md [hunk 6]
```diff
@@ -15,13 +15,13 @@ The 3 frameworks that are supported are:
 
 ## Commonalities
 
-Whatever the framework you pick, your model must be able to accept batches of states, so 2-Dimensonal matrices {code}`(B,N)` where $N$ is the number of local degrees of freedom in the hilbert space (spatial sites) and $B$ is the number of batches.
+Whatever the framework you pick, your model must be able to accept batches of states, so 2-Dimensional matrices {code}`(B,N)` where $N$ is the number of local degrees of freedom in the hilbert space (spatial sites) and $B$ is the number of batches.
```
Comment (femtobit) [inline docs/advanced/custom_models.md:18]:
```suggestion
Whatever the framework you pick, your model must be able to accept batches of states, so 2-dimensional matrices {code}`(B,N)` where $N$ is the number of local degrees of freedom in the hilbert space (spatial sites) and $B$ is the number of batches.
```

**Review by PhilipVinc — APPROVED** [review]
My god. I really need a new laptop that does not double all the e(s) I type.

Thanks a lot!

**Review by femtobit — CHANGES_REQUESTED** [review]
Very comprehensive, thanks!

There are a few cases were you wrote "Änsatze" instead of "Ansätze" (please check whether I got all of them), if you fix those I'm happy. 😁

**Comment by nikosavola** [comment]
> Very comprehensive, thanks!
> 
> There are a few cases were you wrote "Änsatze" instead of "Ansätze" (please check whether I got all of them), if you fix those I'm happy. 😁

There was still one in `qgt_jacobian_dense.py`, but I amended it to the suggestion commit now.

**Review by femtobit — APPROVED** [review]
Thanks!

**Pull request merged**